src/engine/gnc-pricedb.h

```diff
@@ -372,17 +372,29 @@ gboolean     gnc_pricedb_add_price(GNCPriceDB *db, GNCPrice *p);
  */
 gboolean     gnc_pricedb_remove_price(GNCPriceDB *db, GNCPrice *p);
 
+typedef enum
+{
+    PRICE_REMOVE_DEFAULT, // default option value
+    PRICE_REMOVE_USER,    // only user prices
+    PRICE_REMOVE_ALL,     // all prices
+    PRICE_REMOVE_WEEKLY,  // leave one every week on a friday
+    PRICE_REMOVE_MONTHLY, // leave one every last friday of month
+    PRICE_REMOVE_SCALED,  // leave one every week then one a month
+} PriceRemoveOptions;
```

Comment:
Like a mentioned in my other comment, it's confusing to mix different option sets in one enum. Please separate this into two, one for the options to indicate which sources to modify and one for the options to specify which dates to keep.

An example of the confusion: when used in the context of sources to clean up PRICE_REMOVE_DEFAULT means not user source and not all sources. So that would be fq source. However in the context of the "leave" option PRICE_REMOVE_DEFAULT means remove all dates (within the selected sources that is). Those are very different meanings.